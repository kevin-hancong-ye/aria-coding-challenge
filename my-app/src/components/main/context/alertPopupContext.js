import { createContext } from "react";

export const AlertPopupContext = createContext();
