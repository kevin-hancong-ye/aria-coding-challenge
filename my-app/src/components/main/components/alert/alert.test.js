import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { cleanup, render, screen } from "@testing-library/react";
import { AlertPopupContext } from "../../context/alertPopupContext";

import Alert from "./alert";

afterEach(() => cleanup());

test("renders alert component", () => {
	const showAlertPopup = false;
	render(
		<AlertPopupContext.Provider value={[showAlertPopup, () => {}]}>
			<Alert />
		</AlertPopupContext.Provider>
	);

	const alertComponentElement = screen.getByTestId("alert-dialog");
	expect(alertComponentElement).toBeInTheDocument();

	const headingElem = screen.getByRole("heading");
	expect(headingElem).toHaveTextContent("Alert Dialog");
});

test("render alert dialog on press trigger alert button show", async () => {
	const showAlertPopup = true;
	render(
		<AlertPopupContext.Provider value={[showAlertPopup, () => {}]}>
			<Alert />
		</AlertPopupContext.Provider>
	);

	const alertDialogElem = screen.getByTestId("alert-diaglog-info");

	expect(alertDialogElem).toBeInTheDocument();
	expect(alertDialogElem).toMatchSnapshot();
});
