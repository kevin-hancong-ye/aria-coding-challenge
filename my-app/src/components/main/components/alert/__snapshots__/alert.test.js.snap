// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`on press trigger alert button 1`] = `
<div
  data-testid="alert-diaglog-info"
  id="alertText"
>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quis istum dolorem timet? Erat enim res aperta. Habes, inquam, Cato, formam eorum, de quibus loquor, philosophorum. Respondeat totidem verbis. Sin tantum modo ad indicia veteris memoriae cognoscenda, curiosorum. Duo Reges: constructio interrete. Cupit enim dícere nihil posse ad beatam vitam deesse sapienti.
</div>
`;

exports[`render alert dialog on press trigger alert button show 1`] = `
<div
  data-testid="alert-diaglog-info"
  id="alertText"
>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quis istum dolorem timet? Erat enim res aperta. Habes, inquam, Cato, formam eorum, de quibus loquor, philosophorum. Respondeat totidem verbis. Sin tantum modo ad indicia veteris memoriae cognoscenda, curiosorum. Duo Reges: constructio interrete. Cupit enim dícere nihil posse ad beatam vitam deesse sapienti.
</div>
`;
